### PR TITLE
Calculate depth from patient not contact. Issue: #2648

### PR DESCRIPTION
**Pull request by garethbowen**

### CODE REVIEW

File: lib/views.js [hunk 1]
```diff
@@ -266,6 +266,63 @@ exports.places_by_type_parent_id_name = {
...
+      case 'data_record':
+        var patientId = doc.patient_id || (doc.fields && doc.fields.patient_id);
+        if (patientId) {
+          return emit(patientId);
+        }
+        var placeId = doc.place_id || (doc.fields && doc.fields.place_id);
+        if (placeId) {
+          return emit(placeId);
+        }
+        var contact = doc.contact || // incoming message
```
Comment (SCdF) [inline lib/views.js]: Not every `data_record` with a `contact` an incoming message.

Comment (SCdF) [inline lib/views.js]: Oh ignore me, I didn't realise that we're returning the emits higher up. So by now I guess all remaining docs are messages?

File: lib/views.js [hunk 2]
```diff
@@ -266,6 +266,63 @@ exports.places_by_type_parent_id_name = {
...
+};
+
+// WARNING: If updating this function also update the getReplicationKey function in api/handlers/changes.js
+exports.docs_by_replication_key = {
+  map: function(doc) {
+    if (doc._id === 'resources' ||
+        doc._id === 'appcache' ||
+        doc.type === 'form' ||
+        doc.type === 'translations') {
+      return emit('_all');
```
Comment (SCdF) [inline lib/views.js]: would it make sense to add `_design/medic-client` as an id check here and get rid of the hard-coding in API?

File: lib/views.js (hunk 1 above)
Comment (garethbowen) [inline lib/views.js]: Probably...  Try the patient_id, then the place_id, then the contact._id, then give up and call it unassigned.

File: lib/views.js (hunk 2 above)
Comment (garethbowen) [inline lib/views.js]: Yeah good call. Adding.

**Pull request merged**